src/components/ProfilePicture.tsx

```diff
@@ -1,24 +1,39 @@
+import {User} from "next-auth";
+import Link from "next/link";
+import {userPathToProfile} from "~/utils/profile";
+
 interface ProfilePictureProps extends React.ComponentProps<"img"> {
-  imgUrl?: string | null;
   size: number;
+  imgUrl?: string | null;
+  userId?: string;
 }
 /**
  * @prop size: Width/Height of the Profile Picture in REM
  * @returns
  */
-export const ProfilePicture: React.FC<ProfilePictureProps> = ({
-  imgUrl,
-  size,
-  ...props
-}) => {
-  const { className, ...restProps } = props;
-  return (
-    <img
-      src={imgUrl || "https://i.stack.imgur.com/34AD2.jpg"}
-      alt="avatar"
-      className={`inline-block aspect-square rounded-[50%] ${className ?? ""}`}
-      style={{ width: `${size}rem`, height: `${size}rem` }}
-      {...restProps}
-    />
-  );
-};
+
+export const ProfilePicture: React.FC<ProfilePictureProps> = ({ size, imgUrl, userId, ...props }) => {
+    const { className, ...restProps } = props;
+
+    const linkTo: string = userPathToProfile(userId || '');
+    let imageUrl = "https://i.stack.imgur.com/34AD2.jpg";
+
+    if (imgUrl) {
+        imageUrl = imgUrl;
+    }
+
+    const img = <img 
+                    src={imageUrl} 
+                    alt="avatar"
+                    className={`inline-block aspect-square rounded-[50%] ${className ?? ""}`}
+                    style={{ width: `${size}rem` }}
+                    {...restProps}
+/>;
+
+    return (
+        <> {(userId) ?
+            <Link href={linkTo}> img </Link> :
```

Comment:
I'm not sure if this will work. I pulled your changes locally and I see this for all the images.

![image](https://user-images.githubusercontent.com/46171819/227744805-d7e04d14-9ffe-43b2-9b6b-fd867286ae20.png)

Maybe try this instead

```jsx
<Link href={linkTo}>{img}</Link>
```

Reply:
your suggestion was the solution